roles/openshift_version/tasks/first_master_rpm_version.yml

```diff
@@ -6,16 +6,20 @@
   when:
   - openshift_pkg_version is defined
   - openshift_pkg_version != ""
-  - openshift_version is not defined
-  - not (openshift_version_reinit | default(false))
+  - openshift_version is not defined or openshift_version_reinit | bool
 
 # These tasks should only be run against masters and nodes
 - name: Set openshift_version for rpm installation
   include_tasks: check_available_rpms.yml
 
-- set_fact:
+- name: Set openshift_version to rpm version found
+  set_fact:
     openshift_version: "{{ rpm_results.results.versions.available_versions.0 }}"
-  when: openshift_version is not defined or ( openshift_version_reinit | default(false) )
-- set_fact:
+  when:
+  - openshift_version is not defined or openshift_version_reinit | bool
+
+- name: Set openshift_pkg_version to rpm version found if openshift_pkg_version was not specified
+  set_fact:
     openshift_pkg_version: "-{{ rpm_results.results.versions.available_versions.0 }}"
-  when: openshift_version_reinit | default(false)
+  when:
+  - openshift_pkg_version is not defined or openshift_version_reinit | bool
```

Comment:
This will result in incorrect openshift_pkg_version if the user has specified it during 3.9 portion.
Probably should set an additional fact after line 5 to denote we have explicitly set openshift_version to openshift_pkg_version and check not defined that fact here.